graph/src/blockchain/block_stream.rs

```diff
@@ -173,9 +173,22 @@ where
 }
 
 impl<C: Blockchain> BlockWithTriggers<C> {
-    pub fn new(block: C::Block, mut trigger_data: Vec<C::TriggerData>) -> Self {
+    pub fn new(block: C::Block, mut trigger_data: Vec<C::TriggerData>, logger: &Logger) -> Self {
```

Comment:
This to me sounds like a lot of logic for a `new` function. I'd rather change this to something from_trigger_data or something with a comment saying it will dedup etc.

Reply:
I just added a doc comment, what do you think?

Reply:
I don't feel strongly about it but I feel like reading the code you would not expect new to do the sort + dedup etc. The doc helps though I'll leave the rest up to you.